src/Hazelcast.Net/Serialization/Compact/CompactDictionaryGenericRecord.GeneratedCode.cs

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#nullable enable
+
+using Hazelcast.Models;
+
+namespace Hazelcast.Serialization.Compact;
+
+internal partial class CompactDictionaryGenericRecord
+{
+    // <generated>
+
+    /// <inheritdoc />
+    public bool GetBoolean(string fieldname)
+    {
+        ValidateField(fieldname, FieldKind.Boolean, FieldKind.NullableBoolean);
+        return _fieldValues[fieldname] is bool value ? value : throw new SerializationException($"Null value for field '{fieldname}'.");
```

Comment:
Is `SerializationException` a common behavior? I couldn't find a counterpart on Java side.

Reply:
Mmm... there's `HazelcastSerializationException` in Java that is thrown "when a problem occurs while serializing/deserializing objects" and we're going the same in .NET.